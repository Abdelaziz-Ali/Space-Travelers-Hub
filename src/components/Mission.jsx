import React from 'react';
import PropTypes from 'prop-types';
import { useDispatch } from 'react-redux';
import { join, leave } from '../redux/missions/missionSlice';

export default function Mission({
  name, description, status, id,
}) {
  const dispatch = useDispatch();
  return (
    <tr className={status ? 'table-success' : 'table-body'}>
      <td>{name}</td>
      <td>{description}</td>
      <td>
        <span className="badge text-bg-primary">Active member</span>
        <span className="badge text-bg-secondary">Not a member</span>
      </td>
      <td className="text-center col-2">
        {status ? (
          <button
            type="button"
            className="btn btn-outline-danger"
            onClick={() => dispatch(leave(id))}
          >
            Leave Mission
          </button>
        ) : (
          <button
            type="button"
            className="btn btn-outline-secondary"
            onClick={() => dispatch(join(id))}
          >
            Join Mission
          </button>
        )}
      </td>
    </tr>
  );
}

Mission.propTypes = {
  name: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  status: PropTypes.bool.isRequired,
  id: PropTypes.string.isRequired,
};
