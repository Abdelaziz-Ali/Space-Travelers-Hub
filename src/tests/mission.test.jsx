import { Provider } from 'react-redux';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import store from '../redux/store';
import Mission from '../components/Mission';

describe('Tests for mission component', () => {
  const mission = (
    <Provider store={store}>
      <Mission description="Mission 1" id="1" name="my mission" status={false} />
    </Provider>
  );
  test('Take snapshot form the element', () => {
    const component = render(mission);
    expect(component).toMatchSnapshot();
  });

  test('It has a table element', () => {
    render(mission);
    const row = screen.getByTestId('row');
    expect(row).toBeInTheDocument();
  });

  test('Each mission has 4 columns', () => {
    render(mission);
    const row = screen.getByTestId('row');
    expect(row.children).toHaveLength(4);
  });
});
