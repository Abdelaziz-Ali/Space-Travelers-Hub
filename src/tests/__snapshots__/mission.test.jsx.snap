// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for mission component Take snapshot form the element 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <tr
        class="table-body"
        data-testid="row"
      >
        <td>
          my mission
        </td>
        <td>
          Mission 1
        </td>
        <td>
          <span
            class="badge text-bg-secondary"
          >
            Not a member
          </span>
        </td>
        <td
          class="text-center col-2"
        >
          <button
            class="btn btn-outline-secondary"
            type="button"
          >
            Join Mission
          </button>
        </td>
      </tr>
    </div>
  </body>,
  "container": <div>
    <tr
      class="table-body"
      data-testid="row"
    >
      <td>
        my mission
      </td>
      <td>
        Mission 1
      </td>
      <td>
        <span
          class="badge text-bg-secondary"
        >
          Not a member
        </span>
      </td>
      <td
        class="text-center col-2"
      >
        <button
          class="btn btn-outline-secondary"
          type="button"
        >
          Join Mission
        </button>
      </td>
    </tr>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
